python/cudf/cudf/core/reshape.py

```diff
@@ -282,39 +315,40 @@ def concat(objs, axis=0, join="outer", ignore_index=False, sort=None):
         else:
             if axis == 0:
                 result = obj.copy()
+                if keys is not None:
+                    raise NotImplementedError(
+                        "Concatenation along axis = 0 "
+                        "when passing a dictionary is not supported yet."
+                    )
```

Comment:
My suggestion above is to move this check earlier. Rationale is partly that if we were to introduce support for the `keys` argument separately then the check `keys is not None` is not the correct canary for "dictionary input". WDYT?

Reply:
sounds good -- FYI the intention is to fast follow this PR with `axis=0` support